python/vyos/ifconfig/dhcp.py

```diff
@@ -16,103 +16,55 @@
-    send vendor-class-identifier "{{ vendor_class_id }}";
-    {% endif -%}
-    request subnet-mask, broadcast-address, routers, domain-name-servers,
-        rfc3442-classless-static-routes, domain-name, interface-mtu;
-    require subnet-mask;
-}
-
-"""
-
-template_v6 = """
-# generated by ifconfig.py
-interface "{{ intf }}" {
-    request routers, domain-name-servers, domain-name;
-}
-
-"""
-
-class DHCP (Control):
+
+class _DHCP (Control):
     client_base = r'/var/lib/dhcp/dhclient_'
 
-    def __init__ (self, ifname, **kargs):
+    def __init__(self, ifname, version, **kargs):
         super().__init__(**kargs)
-
-        # per interface DHCP config files
-        self._dhcp = {
-            4: {
-                'ifname': ifname,
-                'conf': self.client_base + ifname + '.conf',
-                'pid':  self.client_base + ifname + '.pid',
-                'lease': self.client_base + ifname + '.leases',
-                'options': {
-                    'intf': ifname,
-                    'hostname': '',
-                    'client_id': '',
-                    'vendor_class_id': ''
-                },
-            },
-            6: {
-                'ifname': ifname,
-                'conf': self.client_base + ifname + '.v6conf',
-                'pid':  self.client_base + ifname + '.v6pid',
-                'lease': self.client_base + ifname + '.v6leases',
-                'accept_ra': f'/proc/sys/net/ipv6/conf/{ifname}/accept_ra',
-                'options': {
-                    'intf': ifname,
-                    'dhcpv6_prm_only': False,
-                    'dhcpv6_temporary': False
-                },
-            },
+        self.version = version
+        self.file = {
+            'ifname': ifname,
+            'conf': self.client_base + ifname + '.' + version + 'conf',
+            'pid': self.client_base + ifname + '.' + version + 'pid',
+            'lease': self.client_base + ifname + '.' + version + 'leases',
         }
 
-    def get_dhcp_options(self):
-        """
-        Return dictionary with supported DHCP options.
-
-        Dictionary should be altered and send back via set_dhcp_options()
-        so those options are applied when DHCP is run.
-        """
-        return self._dhcp[4]['options']
-
-    def set_dhcp_options(self, options):
-        """
-        Store new DHCP options used by next run of DHCP client.
-        """
-        self._dhcp[4]['options'] = options
-
-    def get_dhcpv6_options(self):
-        """
-        Return dictionary with supported DHCPv6 options.
-
-        Dictionary should be altered and send back via set_dhcp_options()
-        so those options are applied when DHCP is run.
-        """
-        return self._dhcp[6]['options']
-
-    def set_dhcpv6_options(self, options):
-        """
-        Store new DHCP options used by next run of DHCP client.
-        """
-        self._dhcp[6]['options'] = options
+class _DHCPv4 (_DHCP):
+    template = """\
```

Comment:
Please migrate the template to data/templates folder

Reply:
Hi Christian, I agree it should be done but could you please merge as the current code does not load them from there neither so it is not a step backward.

Reply:
Can‘t as there are conflicting files.

Reply:
argh ! ok - will re,re,...base :-) and add the template at the same time.